### ISSUE TITLE
E2E: Add some structure around namespaces in  E2Es.

### ISSUE THREAD

**Issue by jayunit100** [GoogleCloudPlatform/kubernetes#6554]
TL;DR , see https://github.com/GoogleCloudPlatform/kubernetes/pull/6553  ... justification follows...

The E2E's have many ways of creating mocked up namespaces for tests w/ uids and so on.  

Rather than each test creating a totally ad hoc new namespace, which we have no control over, lets add a top level **--namespace** option.  This will make analyzing etcd, customizing where tests run, and so on, much simpler.

example: there are two different ways which a test can use the `--namespace=hello123` flag.

- Explicit : It only launches tests in hello123 namespace
- Hint : It creates all namespaces based off of hello123 (i.e. hello123-0, hello123-1).

**Comment by jayunit100** [comment]
Closing as we have now tidied this up
